model/host/hostutil.go

```diff
@@ -482,13 +493,40 @@ func (h *Host) BootstrapScript(settings *evergreen.Settings, creds *certdepot.Cr
 		return "", errors.Wrap(err, "could not get commands to write Jasper credentials file")
 	}
 
+	var setupJasperCmds []string
+	setupJasperCmds = append(setupJasperCmds, makeJasperDirs)
+	setupJasperCmds = append(setupJasperCmds, writeCredentialsCmds)
+	setupJasperCmds = append(setupJasperCmds,
+		h.FetchJasperCommand(settings.HostJasper),
+		h.ForceReinstallJasperCommand(settings),
+		fixJasperDirsOwner,
+	)
+
 	bashCmds := append(bashPrefix, checkUserDataRan, setupScriptCmds)
-	bashCmds = append(bashCmds, writeCredentialsCmds, h.FetchJasperCommand(settings.HostJasper), h.ForceReinstallJasperCommand(settings))
+	bashCmds = append(bashCmds, setupJasperCmds...)
 	bashCmds = append(bashCmds, fetchClient, fixClientOwner, postFetchClient, markDone)
 
 	return strings.Join(append([]string{"#!/bin/bash"}, bashCmds...), "\n"), nil
 }
 
+// changeJasperDirsOwnerCommand returns the command to ensure that the Jasper
+// directories have proper permissions.
+func (h *Host) ChangeJasperDirsOwnerCommand() string {
+	return strings.Join([]string{
+		h.changeOwnerCommand(h.Distro.BootstrapSettings.JasperBinaryDir),
+		h.changeOwnerCommand(h.Distro.BootstrapSettings.JasperCredentialsPath),
+		h.changeOwnerCommand(h.Distro.BootstrapSettings.ClientDir),
+	}, " && ")
+}
+
+func (h *Host) MakeJasperDirsCommand() string {
+	return strings.Join([]string{
+		fmt.Sprintf("mkdir -m 777 -p %s", h.Distro.BootstrapSettings.JasperBinaryDir),
+		fmt.Sprintf("mkdir -m 777 -p %s", filepath.Dir(h.Distro.BootstrapSettings.JasperCredentialsPath)),
+		fmt.Sprintf("mkdir -m 777 -p %s", h.Distro.BootstrapSettings.JasperBinaryDir),
+	}, " && ")
```

Comment:
I think you can just `mkdir <IFS specified list of directories>`

Reply:
Done.